src/common/state/table.c

```diff
@@ -101,8 +96,11 @@ int64_t table_timeout_handler(event_loop *loop,
     return EVENT_LOOP_TIMER_DONE;
   }
 
-  /* Decrement retry count and try again. */
-  callback_data->retry.num_retries--;
+  /* Decrement retry count and try again. We use -1 to indicate infinite
+   * retries. */
+  if (callback_data->retry.num_retries != -1) {
```

Comment:
I think I would prefer if we could do exponential backoff for infinite retries, instead of the same timeout every time. Most likely, table callbacks are likely to fail when there's a burst of requests. If we retry them all exactly 10 seconds later, we'll just get the same number of failures the next time around.